packages/build-utils/src/clone-env.ts

```diff
@@ -1,16 +1,29 @@
+import type { ProcessEnv } from './types';
+
 /**
  * Performs a shallow clone on the environment variable object.
  *
  * @param {Object} [currentEnv] The environment object to clone
  * @returns {Object}
  */
-export function cloneEnv(currentEnv = process.env): typeof process.env {
-  return {
+export function cloneEnv(...envs: (ProcessEnv | undefined)[]): ProcessEnv {
+  if (!envs.length) {
+    envs.push(process.env);
+  }
+
+  return envs.reduce((obj: ProcessEnv, env) => {
+    if (env === undefined || env === null) {
+      return obj;
+    }
+
     // the system path is called `Path` on Windows and Node.js will
     // automatically return the system path when accessing `PATH`,
     // however we lose this proxied value when we destructure and
     // thus we must explicitly copy it
-    PATH: currentEnv.PATH,
-    ...currentEnv,
-  };
+    if (obj.PATH === undefined) {
```

Comment:
Not sure this is correct. 

I would expect `if (env.PATH !== undefined))`

Reply:
Gah, coding too fast.

Reply:
Lets make sure to add a test to cover this case

Reply:
I updated the code such that it will copy `PATH` if the property exists, regardless if it's `undefined`. It might be handy to unset the `PATH` with a simple `PATH: undefined`.